packages/capabilities/src/http.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * HTTP Capabilities.
+ *
+ * These can be imported directly with:
+ * ```js
+ * import * as HTTP from '@web3-storage/capabilities/http'
+ * ```
+ *
+ * @module
+ */
+import { capability, Schema, ok } from '@ucanto/validator'
+import { content } from './blob.js'
+import { equal, equalBody, equalWith, SpaceDID, and } from './utils.js'
+
+/**
+ * `http/put` capability invocation MAY be performed by any agent on behalf of the subject.
+ * The `blob/add` provider MUST add `/http/put` effect and capture private key of the
+ * `subject` in the `meta` field so that any agent could perform it.
+ */
+export const put = capability({
+  can: 'http/put',
+  /**
+   * DID of the (memory) space where Blob is intended to
+   * be stored.
+   */
+  with: SpaceDID,
+  nb: Schema.struct({
+    /**
+     * BodyBlob to allocate on the space.
+     */
+    body: content,
```

Comment:
Personally I'd just repeat the fields here so this doesn't depend on blob capability.

Reply:
probably better to not tie the implementation with this, this follows the spec, so I would recommend to discuss this at the spec level via issue/PR rather than here... see https://github.com/web3-storage/specs/pull/118

Reply:
Moved this to https://github.com/web3-storage/specs/pull/118/files#r1558328467